src/test/java/bio/terra/janitor/app/controller/JanitorApiControllerTest.java

```diff
@@ -0,0 +1,75 @@
+package bio.terra.janitor.app.controller;
+
+import static bio.terra.janitor.app.common.TestUtils.*;
+import static bio.terra.janitor.app.common.TestUtils.DEFAULT_LABELS;
+import static bio.terra.janitor.common.ResourceType.GOOGLE_PROJECT;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import bio.terra.generated.model.CloudResourceUid;
+import bio.terra.janitor.app.Main;
+import bio.terra.janitor.app.configuration.JanitorJdbcConfiguration;
+import java.util.Optional;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.result.MockMvcResultHandlers;
+
+@Tag("unit")
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = Main.class)
+@SpringBootTest
+@AutoConfigureMockMvc
+public class JanitorApiControllerTest {
+  @Autowired private MockMvc mvc;
+  @Autowired JanitorJdbcConfiguration jdbcConfiguration;
+
+  private NamedParameterJdbcTemplate jdbcTemplate;
+
+  @BeforeEach
+  public void setup() {
+    jdbcTemplate = new NamedParameterJdbcTemplate(jdbcConfiguration.getDataSource());
+  }
+
+  @Test
+  public void testCreateResourceSuccess() throws Exception {
+    CloudResourceUid cloudResourceUid = newGoogleProjectResourceUid();
+    this.mvc
+        .perform(
+            post("/api/janitor/v1/resource")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(newJsonCreateRequestBody(cloudResourceUid, Optional.of(DEFAULT_LABELS))))
+        .andDo(MockMvcResultHandlers.print())
+        .andExpect(status().isOk());
+
+    // Verify data are created correctly.
+    assertCreateResultMatch(
+        cloudResourceUid,
+        GOOGLE_PROJECT,
+        Optional.empty(),
+        Optional.empty(),
+        TIME_TO_LIVE_MINUTE,
+        jdbcTemplate,
+        DEFAULT_LABELS);
```

Comment:
I don't get why some of the TestUtils private static final variables are implicit in these functions, but only some of the time. e.g. TIME_TO_LIVE_MINUTE is inferred in newJsonCreateRequestBody, but is explicitly passed into assertCreateResultMatch. GOOGLE_PROJECT is used by newGoogleProjectResourceUid, but passed into assertCreateResultMatch. This makes it really hard to tell what's coming from where. I think the reason things are getting pulled into test helpers is because this is testing the same thing as the JanitorDao using the same methods. I think that's too much internal looking by this test.
Make things more explicit, or consistently implicit. Think if there's a test we can write that's testing less of the JanitorDao's implementation. Do less in TestUtils.

Reply:
Done, per discussed offline, assert db record match will be the DaoTest's job. I spent quite a lot time trying to make a code reusable as much as possible.  Then the code looks confusion. Updated a bit, PTAL